src/utils/props.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * project any angle onto the interval (-180;180]
+ */
+export function normalizeAngle(x: number) {
+  while (x > 0) x -= 360;
+  return (x - 180) % 360 + 180;
+}

```

Comment:
it's considered a bad practice to reassign arguments' values. please, define a separate mutable variable for storing `x` instead